examples/Example/src/new_api/calculator/index.tsx

```diff
@@ -0,0 +1,399 @@
+    });
+    dragOffset.value = -props.offset.value;
+
+    setOpened(true);
+  }
+
+  function close() {
+    drag.value = withTiming(0, { duration: DRAG_ANIMATION_DURATION });
+    dragOffset.value = 0;
+
+    setOpened(false);
+  }
+
+  const translationStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: props.offset.value + drag.value }],
+    };
+  });
+
+  const dragGesture = useAnimatedGesture(
+    Gesture.pan()
+      .setOnUpdate((e) => {
+        'worklet';
+        const value = dragOffset.value + e.translationY;
+
+        if (value > -props.offset.value) {
+          drag.value = -props.offset.value;
+        } else if (value < 0) {
+          drag.value = 0;
+        } else {
+          drag.value = value;
+        }
+      })
+      .setOnEnd((e) => {
+        'worklet';
+        const value = dragOffset.value + e.translationY;
+
+        if (opened) {
+          if (value < -props.offset.value - 100) {
+            runOnJS(close)();
+          } else {
+            runOnJS(open)();
+          }
+        } else {
+          if (value > 100) {
+            runOnJS(open)();
+          } else {
+            runOnJS(close)();
+          }
+        }
+      })
+  );
+
+  scrollView.current?.scrollToEnd({ animated: true });
+
+  return (
+    <GestureMonitor gesture={dragGesture}>
+      <Animated.View
+        style={[styles.output, translationStyle]}
+        onLayout={measure}>
+        <ScrollView
+          ref={(ref: ScrollView) => {
+            if (!opened && ref) ref.scrollToEnd({ animated: false });
+            scrollView.current = ref;
+          }}
+          enabled={opened}
+          contentContainerStyle={{ flexGrow: 1 }}>
+          <View style={{ flexGrow: 1 }} />
+
+          {props.history.map((exp: string) => {
+            return <Expression expression={exp} key={exp} />;
+          })}
+
+          <Expression expression={props.expression} />
+        </ScrollView>
+        <View style={styles.handleView}>
+          <View style={styles.handle} />
+        </View>
+      </Animated.View>
+    </GestureMonitor>
+  );
+}
+
+function Expression(props: { expression: string }) {
+  return (
+    <View style={styles.expression}>
+      <View style={{ flexDirection: 'column' }}>
+        <Text style={styles.expressionText}>{props.expression}</Text>
+        <Text style={styles.expressionResult}>{props.expression}</Text>
+      </View>
+    </View>
+  );
+}
+
+function Input(props: {
+  setHistory: Dispatch<SetStateAction<string[]>>;
+  setExpression: Dispatch<SetStateAction<string>>;
+  measure: (e: LayoutChangeEvent) => void;
+  offset: Animated.SharedValue<number>;
+  expression: string;
+}) {
+  const translationStyle = useAnimatedStyle(() => {
+    return {
+      transform: [{ translateY: props.offset.value }],
+    };
+  });
+
+  function append(symbol: string) {
+    if (symbol === '<') {
+      props.setHistory((h) => [...h, props.expression]);
```

Comment:
`h.concat(expression)`

Reply:
Done in 26c0974724e12b9c9e06ce531674471ac46d05c3